### PR TITLE
8.0 sale order completion

### PR DESCRIPTION
**Pull request by chafique-delli**
add account_statement_sale_order module for create a self-completion method fill:
    command, partner and receivable account of a bank statement line.

### CODE REVIEW

File: account_statement_sale_order/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+##############################################################################
+
+{
+    'name': 'Bank Statement Sale Order',
+    'version': '0.1',
+    'category': 'Finance',
+    'license': 'AGPL-3',
+    'summary': '',
+    'description': """
+    The goal of this module is to create a self-completion method fill:
```
Comment (pedrobaeza) [inline account_statement_sale_order/__openerp__.py]: Extract this to a README.rst file and remove from here. You should explain a bit more the purpose of the module (I didn't know very well yet), and use the OCA's template: https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

File: account_statement_sale_order/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+{
+    'name': 'Bank Statement Sale Order',
+    'version': '0.1',
```
Comment (pedrobaeza) [inline account_statement_sale_order/__openerp__.py]: Use 1.0 as starting version

File: account_statement_sale_order/sale.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+#
+#   You should have received a copy of the GNU Affero General Public License
+#   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+###############################################################################
+
+from openerp.osv import orm
+
+
+class SaleOrder(orm.Model):
```
Comment (pedrobaeza) [inline account_statement_sale_order/sale.py]: You can use new API as this is very short

File: account_statement_sale_order/statement.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    account_statement_sale_order for OpenERP
+#    Copyright (C) 2013-14 Akretion Chafique DELLI <chafique.delli@akretion.com>
```
Comment (pedrobaeza) [inline account_statement_sale_order/statement.py]: PEP8 80 cols limit

File: account_statement_sale_order/statement.py [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.tools.translate import _
+from openerp.osv import orm, fields
+
+
+class AccountStatementLine(orm.Model):
```
Comment (pedrobaeza) [inline account_statement_sale_order/statement.py]: new API for this?

File: account_statement_sale_order/statement.py [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    _inherit = "account.bank.statement.line"
+
+    _columns = {
+        'sale_ids': fields.many2many('sale.order', string='Sale Orders',)
```
Comment (pedrobaeza) [inline account_statement_sale_order/statement.py]: This field is not shown in interface. Maybe it should?

File: account_statement_sale_order/statement.py [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+        (_check_partner_id,
+         'The sale orders chosen have to belong to the same partner',
+         ['sale_ids']),
+    ]
+
+
+class AccountBankStatement(orm.Model):
+    _inherit = "account.bank.statement"
+
+    def balance_check(self, *args, **kwargs):
```
Comment (pedrobaeza) [inline account_statement_sale_order/statement.py]: For what is this needed?

File: account_statement_sale_order/statement.py [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+            ctx = {}
+        else:
+            ctx = context.copy()
+        ctx['balance_check'] = True
+        kwargs['context'] = ctx
+        return super(AccountBankStatement, self).balance_check(
+            *args, **kwargs)
+
+
+    #def _prepare_counterpart_move_line(self, *args, **kwargs):
```
Comment (pedrobaeza) [inline account_statement_sale_order/statement.py]: Remove this

**Comment by pedrobaeza** [comment]
The module name should be _account_bank_statement_sale_order_, to match the rest of the names about this area.

File: account_bank_statement_sale_order/statement.py [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+            mv_line_dicts[0]['sale_ids'] = [
+                (6, 0, [sale.id for sale in self.sale_ids])
+            ]
+        super(AccountStatementLine, self).process_reconciliation(mv_line_dicts)
+
+
+class AccountBankStatement(models.Model):
+    _inherit = 'account.bank.statement'
+
+    @api.model
```
Comment (sebastienbeau) [inline account_bank_statement_sale_order/statement.py:78]: Your last commit broke the module. Indeed in combinaison with the pos_sale_order I can not close any more the session.
Maybe removing _arg, *_kawrgs will fix the issue

**Comment by legalsylvain** [comment]
@sebastienbeau : Is it now ok ? I'd like to review https://github.com/OCA/pos/pull/35 that depends on this PR.
kind regards.

**Comment by rafaelbn** [comment]
Hi @sebastienbeau @chafique-delli please could you review this PR in order to get it merged? As @legalsylvain it's needed. Thanks!

**Comment by pedrobaeza** [comment]
No answer, so closing. Feel free to reopen if needed.

**Pull request closed without merging**